furnace.c

```diff
@@ -227,14 +248,103 @@ set_max_pwm_safe(furnace_context_t *ctx, int new_max_pwm)
 static void
 command_handler(furnace_context_t* ctx, uint8_t* buffer, void (*feedback)(const char*, const size_t))
 {
-  unsigned arg;
+  unsigned arg, arg2;
   char     str_arg[BUF_SIZE];
 
   if (buffer[0] == '\n') return;
 
   if (memcmp(buffer, "reboot", 6) == 0) {
     reset_usb_boot(0,0);
-  } else if (sscanf(buffer, "max_pwm %u", &arg) == 1) {
+  } 
+#if CONFIG_SPIN_COATER
+#if CONFIG_SPIN_COATER_PWM
+  else if (sscanf((const char*)buffer, "spin_pwm %u", &arg) == 1) {
+      const bool res = set_spin_coater_pwm_safe(ctx, arg);
+      if (!res) {
+        DEBUG_printf("Chaning PWM duty failed\n");
+        return;
+      }
+      if (arg == PWM_IDLE_DUTY) {
+        ctx->spin_coater.spin_state = SPIN_IDLE;
+        cancel_alarm(ctx->spin_coater.timer_id);
+        const char* msg = "spin_stopped\n";
+        feedback(msg, strlen(msg));
```

Comment:
Hmm.. We would normally do:
```C
const char msg[] = "spin_stopped\r\n";
const size_t msg_len = sizeof(msg)-1;
feedback(msr, msg_len);
```
Although, it's interesting which one generates smaller code. Our approach definitely generate more instructions and use stack to store the string, but (hopefully) GCC doesn't allocate anything in `.rodata` for it. I can't say which is better, so I don't care for now.

Reply:
I've aligned all places with your approach so the code is consequent in all places